plugin.lua

```diff
@@ -1,185 +1,151 @@
-            local bpm = tp.Bpm
-            local mspb = 60000/bpm
+    _, vars.selectedOnly = imgui.Checkbox("Use selected notes only",
+                                          vars.selectedOnly)
 
-            local mspcheck1 = mspb / n1
-            local mspcheck2 = mspb / n2
+    imgui.PopItemWidth()
 
-            local checktime1 = starttime
-            local checktime2 = starttime
+    imgui.Dummy({0, 5})
 
-            while endtime > notes[i].StartTime do
-                while checktime1 < notes[i].StartTime do
-                    checktime1 = checktime1 + mspcheck1
-                end
-                while checktime2 < notes[i].StartTime do
-                    checktime2 = checktime2 + mspcheck2
-                end
-
-                deviance1 = notes[i].StartTime - math.floor(checktime1)
-                deviance2 = notes[i].StartTime - math.floor(checktime2)
-
-                --debug = not ((math.abs(deviance1) <= leniency) or (math.abs(mspcheck1 - deviance1) <= leniency) or (math.abs(deviance2) <= leniency) or (math.abs(mspcheck2 - deviance2) <= leniency))
-
-                if not ((math.abs(deviance1) <= leniency) or (math.abs(mspcheck1 - deviance1) <= leniency) or (math.abs(deviance2) <= leniency) or (math.abs(mspcheck2 - deviance2) <= leniency)) then
-                    table.insert(errors, notes[i])
-                end
-
-                if i < #notes then
-                    i = i + 1
-                else
-                    break
-                end
-            end
+    if imgui.Button("Find", {ITEM_WIDTH, 24}) then
+        local snaps = {}
+        for snap in string.gmatch(vars.snaps, "%d+") do
+            table.insert(snaps, tonumber(snap))
         end
-
-        currentobject = 0
-        errorstring = ""
-
-        for _,note in pairs(errors) do
-            errorstring = errorstring .. note.StartTime .. "|" .. note.Lane .. ", "
-        end
-        errorstring = errorstring:sub(1,-3)
-
-        if errorstring == "" then
-            errorstring = "No unsnapped notes detected."
-        end
-
-        showcopytext = false
+        local notes = vars.selectedOnly and state.SelectedHitObjects or
+                          map.HitObjects
+        vars.deltaInfo = findAllDeltas(snaps, notes, vars.leniency)
     end
 
-    imgui.SameLine(0, 4)
-    if imgui.Button("Detect 1 ms Differences Between Notes") then
-        local notes = map.HitObjects
-        errors = {}
-        for i = 2, #notes do
-            if notes[i].StartTime - notes[i-1].StartTime == 1 then
-                table.insert(errors, notes[i-1])
-            end
-        end
+    imgui.Dummy({0, 5})
 
-        currentobject = 0
-        errorstring = ""
-
-        for _,note in pairs(errors) do
-            errorstring = errorstring .. note.StartTime .. "|" .. note.Lane .. ", "
-        end
-        errorstring = errorstring:sub(1,-3)
+    if #vars.deltaInfo == 0 then
+        imgui.Text("No notes to resnap")
```

Comment:
the "No Notes to resnap" message should start off hidden so that the user has a confirmation that the plugin found no unsnapped notes when button is clicked